### ISSUE TITLE
[BUG] Equal height and weight keymap is not working with toggle term window

### ISSUE THREAD

**Issue by Demianeen** [akinsho/toggleterm.nvim#550]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When the toggle term terminal is open, the equal height and weight keymap doesn't do anything.

https://github.com/akinsho/toggleterm.nvim/assets/51330172/2cb9d246-4c19-4427-b040-81d051650a2e

### Expected Behavior

The height/width would become equal across windows

### Steps To Reproduce

1. With environment below
2. Use zero config, setup like:
```lua
return {
  'akinsho/toggleterm.nvim',
  opts = {},
}
```
3. Open toggleterm with `:Toggleterm`
4. Try to use <C-w>= (equal height/width across terminals)
5. See that nothing happens
6. Try to use any other keymap like max height (<C-w>_) or increase decrease width
7. See that they are working

### Environment

```Markdown
- OS: MacOS Sonoma 14.2.1 (23C71)
- neovim version: v0.10.0-dev-2273+g21df0cdb8 RelWithDebInfo 
- LuaJIT 2.1.1706708390
- Shell: zsh
```

**Comment by akinsho** [comment]
Toggleterm *controls* the size of the window using the size parameter so `=` will not work as changing the size must go through toggleterm mechanism to make controlling the windows easier to manage

**Comment by Demianeen** [comment]
> Toggleterm _controls_ the size of the window using the size parameter so `=` will not work as changing the size must go through toggleterm mechanism to make controlling the windows easier to manage

Can you control the size without creating new windows every time? I think it might be useful for those who come with the same problem in the future

**Comment by akinsho** [comment]
It's possible but this isn't something that I designed this for, especially when it comes to tiling terminals in horizontal or vertical directions having a consistent size is useful. As I mention in the README I created this plugin with quite simple/limited functionality in mind. Every bit of increased complexity (which this is despite seeming quite simple) is not worth the maintenance burden for me. As always happy to review well thought out PRs though
